test/refundTransaction.js

```diff
@@ -2,48 +2,381 @@ const $util = require("./util")
 const InkProtocol = artifacts.require("./mocks/InkProtocolMock.sol")
 
 contract("InkProtocol", (accounts) => {
-  let protocol
-
-  beforeEach(async () => {
-    protocol = await InkProtocol.new()
-  })
+  let buyer = accounts[1]
+  let seller = accounts[2]
+  let unknown = accounts[accounts.length - 1]
 
   describe("#refundTransaction()", () => {
-    it("fails for buyer")
-    it("fails for owner")
-    it("fails for mediator")
-    it("fails for policy")
-    it("fails for unknown address")
-    it("fails when transaction does not exist")
+    it("fails for buyer", async () => {
+      let {
+        protocol,
+        transaction
+      } = await $util.buildTransaction(buyer, seller, {
+        finalState: $util.states.Accepted
+      })
+
+      await $util.assertVMExceptionAsync(protocol.refundTransaction(transaction.id, { from: buyer }))
+    })
+
+    it("fails for owner", async () => {
+      let {
+        owner,
+        protocol,
+        transaction
+      } = await $util.buildTransaction(buyer, seller, {
+        finalState: $util.states.Accepted,
+        owner: true
+      })
+
+      await $util.assertVMExceptionAsync(owner.proxyRefundTransaction(protocol.address, transaction.id))
+    })
+
+    it("fails for mediator", async () => {
+      let {
+        mediator,
+        protocol,
+        transaction
+      } = await $util.buildTransaction(buyer, seller, {
+        finalState: $util.states.Accepted
+      })
+
+      await $util.assertVMExceptionAsync(mediator.proxyRefundTransaction(protocol.address, transaction.id))
+    })
+
+    it("fails for policy", async () => {
+      let {
+        policy,
+        protocol,
+        transaction
+      } = await $util.buildTransaction(buyer, seller, {
+        finalState: $util.states.Accepted
+      })
+
+      await $util.assertVMExceptionAsync(policy.proxyRefundTransaction(protocol.address, transaction.id))
+    })
+
+    it("fails for unknown address", async () => {
+      let {
+        protocol,
+        transaction
+      } = await $util.buildTransaction(buyer, seller, {
+        finalState: $util.states.Accepted
+      })
+
+      await $util.assertVMExceptionAsync(protocol.refundTransaction(transaction.id, { from: unknown }))
+    })
+
+    it("fails when transaction does not exist", async () => {
+      let protocol = await InkProtocol.new()
+
+      await $util.assertVMExceptionAsync(protocol.refundTransaction(0, {from: seller}))
```

Comment:
I like what what @kraytruthg did for these tests... he still setup a whole transaction and got it to the proper state. But he called the method with `id + 1` by the user who is able to make the call.

Reply:
I don't really see any benefits from that, since the transaction does not exists, why make a transaction that's not relevant to the test?